tools/edge_cut/src/surf/meshUtils.cpp

```diff
@@ -0,0 +1,293 @@
+ *
+ * @section LICENSE
+ * Copyright (c) 2018, Regents of the University of California
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @section DESCRIPTION
+ * Utilities for managing mesh data structures in edge_cut
+ **/
+
+#include <CGAL/IO/OFF_reader.h>  // Must include here to avoid multiple defn errors (CGAL bug as of v4.13)
+#include "surf/meshUtils.h"
+#include "io/logging.hpp"
+
+edge_cut::surf::K::FT
+edge_cut::surf::SizingField::operator()(  Point_3 const & p,
+                                          int     const,
+                                          Index   const &   ) const
+{
+  FT l_distance = std::sqrt( std::pow( p.x()-m_center.x(), 2 ) + std::pow( p.y()-m_center.y(), 2 ) );
+
+  if ( l_distance <= m_innerRad )
+    return m_innerVal;
+  else if ( l_distance >= m_outerRad )
+    return m_scale * m_innerVal;
+  else {
+    // Control should never reach here if inner and outer radii are equal
+    assert( m_outerRad != m_innerRad );
+    return ( 1 + ( l_distance - m_innerRad ) * ( m_scale - 1) * m_widthInv ) * m_innerVal;
+  }
+}
+
+
+edge_cut::surf::Polyline_type
+edge_cut::surf::topoIntersect(  Poly_slicer & i_slicer,
+                                K::Plane_3    i_plane   )
+{
+  Polylines     polylines;
+  Polyline_type ridges;
+
+  i_slicer( i_plane, std::back_inserter( polylines ) );
+
+  if( polylines.size() == 1 ) {
+    ridges = *(polylines.begin());
+  }
+  else {
+    EDGE_LOG_ERROR << "Error: Polyline slicer did not return precisely one intersection!";
```

Comment:
``EDGE_LOG_ERROR`` does not abort the code, is this the intended behavior.
``EDGE_LOG_FATAL`` does.